### ISSUE TITLE
Thunderbird Version 91.0

### ISSUE THREAD

**Issue by Wileecoyoteig** [thundernest/import-export-tools-ng#260]
Is the current version of ImportExportTools NG 10.0.4 compatible with TB 91?
Is ImportExportTools NG being actively developed?

**Comment by chartre** [comment]
Well, seems that the answer is "not" for both questions.
I had TB 78 this morning and successfuly installed this extension from the official channel (never used before). Tested the export of all _emails and their structure_. Looked fine.
Then I upgraded to the 91 (and 91.0.1 later) and imported the old profile manualy as I was testing if an unrelated bug was solved (it is not 😢). Then, I decided to just do a fresh install with new configuration options and just import all the emails. Guess the result: not installed because is not compatible. And here I am, considering downgrading to 78...

**Comment by cleidigh** [comment]
@chartre 
@Wileecoyoteig 
I am playing serious catch-up,  I lost a month being sick, complletely off-line
currently rushing to get 91 support at least for my primary  extensions
I am working on printiing toolls at the moment, however, I will do
a quick check If immport expport  will partially function
I am also slower as my disability progresses
I will have  to work tto help mainntain 
Opening a status for this
Christopher

**Comment by Wileecoyoteig** [comment]
Christopher,
Thanks for the update.
Please take care of your health first.
I would rather wait and have a fully functioning ImportExportTools NG again.
Thank you.

**Comment by cleidigh** [comment]
@Wileecoyoteig 
Thanks for the support!
hopefully the conversion will not be too bad
Christopher

**Comment by azhe403** [comment]
Thunderbird 92 please

![gambar](https://user-images.githubusercontent.com/17938038/130560723-8ca4f867-cd7f-47dd-8052-017834c769d8.png)

**Comment by mirekgab** [comment]
Hi @cleidigh 
Do you need help with this issue? I did some tests and I think I can help you with this, but I will need some help from you.

**Comment by MysticJay** [comment]
Hi,
export to PDF is the most needed feature at my end. Someone released the installation of ThunderbirdV91 on top of V78 and now I am stuck with a lot of work downgrading some 20 systems back to V78.

A heads-up or a beta would really help here is available anywhere.

I checked @mirekgab 's fork for an XPI, but couldn't find one.

So anything I can do to avoid this nasty downgrade process?

**Comment by cleidigh** [comment]
all
Tract #264

**Comment by cleidigh** [comment]
MysticJay

I am Working it
I have to figure out the new printing engine for  PDF

**Comment by malix0** [comment]
@cleidigh thank you for Your effort and take care of your health.

**Comment by cleidigh** [comment]
@mirekgab
just did the merge
EML inport What does not work is 
Did you see this?
![image](https://user-images.githubusercontent.com/25272315/132399565-d7613010-7bfe-4dcc-b731-6bb1641cd34a.png)

**Comment by mirekgab** [comment]
@cleidigh 
Sorry, my mistake. I've only tested import to local folder, not to imap. I will look at this error.

**Comment by cleidigh** [comment]
I found it, I will Post

**Comment by mirekgab** [comment]
I found probably the same as you
https://developer.thunderbird.net/add-ons/updating/tb91/changes#renamed-or-replaced-api

**Comment by cleidigh** [comment]
exactly

I am updating now

but then I will post b2

but then I will let you continue with EML?

so we do not conflict

sounds good?

**Comment by mirekgab** [comment]
Ok

**Comment by cleidigh** [comment]
The following Partially works

problem is It dels existing  messages
do you want to investigate?

```JavaScript
var cs = Cc["@mozilla.org/messenger/messagecopyservice;1"]
			.getService(Ci.nsIMsgCopyService);
			if (versionChecker.compare(currentVersion, "80") >= 0) {
				cs.copyFileMessage(file, msgFolder, null, false, 1, "", importEMLlistener, msgWindow);
			} else {
				cs.CopyFileMessage(file, msgFolder, null, false, 1, "", importEMLlistener, msgWindow);
			}

**Comment by mirekgab** [comment]
Yes, I can try.

**Comment by mirekgab** [comment]
I tested importing EML files with your patch and my conclusions are as follows:
- importing eml file - works for local and imap folders
- importing all eml files from a folder - works for a local folder, imap has a problem with refreshing the view
- importing all eml files with subfolder - works for a local folder, not works with imap.
Importing with subfolders to imap also doesn't work in v10.0.2 in Thunderbird 78.
I will continue to investigate this issue.

**Comment by jobisoft** [comment]
What repo is the one where TB91 support is developed? This or the one from @mirekgab?

@mirekgab :  I saw that you made changes to the WindowListener API. Please use the latest official version which is compatible with TB91.
https://github.com/thundernest/addon-developer-support/tree/master/wrapper-apis/WindowListener

**Comment by mirekgab** [comment]
@jobisoft 
This repo is one where TB91 support is developed. My repo was created for test purpose only.

**Comment by jobisoft** [comment]
I created a pull request to get most of the functionality back in TB91:
https://github.com/thundernest/import-export-tools-ng/pull/268

Please test.

I have no idea how the backup feature should work, as it was broken in the current released version for TB78 already (for me at least). So that still does not work. Also PDF export is still broken.

**Comment by DavidBE-UK** [comment]
Well, after changing the max version to 93 in the manifest files, beta 2 then beta 3 added themselves to the TB menus and exported a profile for backup purposes in TB 93.0b2. First versions of IET to show themselves in the TB menus, let alone do anything, since I moved to the Thunderbird betas back in June. Thanks for that.

**Comment by jobisoft** [comment]
Closing this issue in favor of https://github.com/thundernest/import-export-tools-ng/issues/264

**Comment by dm17** [comment]
Seems this add on doesn't work on Thunderbird 91.5, does it? The import buttons open and import mail and import everything options which has no options to import mbox files... It still doesn't make sense how to use this add on after reading over the tutorials and the rest. It gives the "no application to import data from found" when clicking next.

**Comment by bmazak** [comment]
Just setup TB 102 and trying to import messages or all from the PST file that has a password on it.  Message from TB "no application found to import from".  Why are these problems still there from Aug 2021 and here we are in July2 2022.  
I tried to make the PST file connected to a pop3 account and still does not find it.  Tried it with IMAP in Outlook 2010, still nothing.  Put the PST file in the default location and still no application found.  I don't like keeping my life info files where everyone knows where to look for them, it's not finding outlook at all.
So, here I am unable to use TB at all. 
As a second question, I have all my accounts using a Local file in TB.  I don't understand MBOX, if I have 8 accounts I only want one inbox, doesn't everybody want that? So, do I have multiple data files? One for each account like IMAP, Why? Nobody wants that!  I selected local file for each account as I added them, wish I could have imported those setting, but I had just converted to IMAP not knowing I could not use a single data file in Outlook 2010.  I guess what I need to know is where, and what should be running, not running outlook instructions for TB Import to work?  Do I have to run outlook to do the import? Do I have to have IMAP to do it? Do I have to have one PST to run Import?  Where are the instructions to do it with all the variable kinds of setups outlook has available, What is TB looking for, will somebody explain that?
To explain, I started to change outlook to convert from one PST file and all accounts pop3, to all IMAP.  Then I figured if I can't have one inbox, I'm going to get rid of Outlook 2010 and use TB.  Now, I'm stuck with TB and no Contacts, no calendar, so, it is useless.  Can someone explain that?  Thank You.
